src/main/java/org/javacs/lsp/MarkedString.java

```diff
@@ -1,12 +1,46 @@
 package org.javacs.lsp;
 
+import java.io.IOException;
+
+import com.google.gson.annotations.JsonAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+import com.google.gson.stream.JsonWriter;
+import com.google.gson.TypeAdapter;
+
+@JsonAdapter(MarkedString.Adapter.class)
 public class MarkedString {
     public String language, value;
 
     public MarkedString() {}
 
+    public MarkedString(String value) {
+        this.value = value;
+    }
+
     public MarkedString(String language, String value) {
         this.language = language;
         this.value = value;
     }
+
+    public static class Adapter extends TypeAdapter<MarkedString> {
+        @Override
+        public void write(JsonWriter out, MarkedString markedString) throws IOException {
+            if (markedString.language == null) {
+                out.value(markedString.value);
+            } else {
+                out.beginObject();
+                out.name("language");
+                out.value(markedString.language);
+                out.name("value");
+                out.value(markedString.value);
+                out.endObject();
+            }
```

Comment:
`MarkedString` is actually a union type, it can be either a plaintext/markdown string, or a codeblock:

>type MarkedString = string | { language: string; value: string };

Ref: https://microsoft.github.io/language-server-protocol/specification#textDocument_hover